src/debug_assert.h

```diff
@@ -23,16 +23,15 @@
  * of our control (e.g., unexpected changes to the metadata) so if you have a
  * test that trigger the error, this macro should not be used.
  */
-#ifdef USE_ASSERT_CHECKING
-#define Ensure(COND, FMT, ...) AssertMacro(COND)
-#else
 #define Ensure(COND, FMT, ...)                                                                     \
 	do                                                                                             \
 	{                                                                                              \
 		if (unlikely(!(COND)))                                                                     \
+		{                                                                                          \
+			Assert(false);                                                                         \
```

Comment:
Is this correct? I mean if the condition is false it is intended to ereport on release builds. Or am I missing something?

Reply:
Assert(false) will only be active in debug builds and trigger coredump on prod builds you will get the error